tfe/organization_members_helpers.go

```diff
@@ -40,3 +41,61 @@ func fetchOrganizationMembers(client *tfe.Client, orgName string) ([]map[string]
 
 	return members, membersWaiting, nil
 }
+
+func fetchOrganizationMemberByNameOrEmail(ctx context.Context, client *tfe.Client, organization, username, email string) (*tfe.OrganizationMembership, error) {
+	if email == "" && username == "" {
+		return nil, fmt.Errorf("you must specify a username or email.")
+	}
+
+	options := &tfe.OrganizationMembershipListOptions{
+		Include: []tfe.OrgMembershipIncludeOpt{tfe.OrgMembershipUser},
+	}
+
+	if email != "" {
+		options.Emails = []string{email}
+	}
+
+	if username != "" {
+		options.Query = username
+	}
+
+	oml, err := client.OrganizationMemberships.List(ctx, organization, options)
+	if err != nil {
+		return nil, fmt.Errorf("failed to list organization memberships: %w", err)
+	}
+
+	switch len(oml.Items) {
+	case 0:
+		return nil, tfe.ErrResourceNotFound
+	case 1:
+		user := oml.Items[0].User
+
+		// We check this just in case a user's TFE instance only has one organization member
+		if user.Email != email && user.Username != username {
+			return nil, tfe.ErrResourceNotFound
+		}
+
+		return oml.Items[0], nil
+	default:
+		for {
+			for _, member := range oml.Items {
+				if member.User.Email == email || member.User.Username == username {
```

Comment:
This works fine, but I would appreciate a little guardrail around this condition to reinforce the relationship with the parameters, in case the username is ever a blank string instead of null.

```suggestion
				if (len(email) > 0 && member.User.Email == email) || (len(username) > 0 && member.User.Username == username) {
```

Reply:
Great suggestion, will add.